src/dmd/cparse.d

```diff
@@ -59,8 +59,16 @@ final class CParser(AST) : Parser!AST
      */
     Array!(void*) typedefTab;  /// Array of AST.Type[Identifier], typedef's indexed by Identifier
 
+    /* This is passed in as a list of #define lines, as generated by the C preprocessor with the
+     * appropriate switch to emit them. We append to it any #define's and #undef's encountered in the source
+     * file, as cpp with the -dD embeds them in the preprocessed output file.
+     * Once the file is parsed, then the #define's are converted to D symbols and appended to the array
+     * of Dsymbols returned by parseModule().
+     */
+    OutBuffer* defines;
```

Comment:
Can this be an associative array?

Reply:
Since it is never searched, it wouldn't help.

Reply:
I meant more in terms documentation. The name and comment isn't as descriptive as I'd like.

Reply:
What is unclear about it? Please be more specific.